### ISSUE TITLE
Split `win_injector.c` into separate files for each injection method

### ISSUE THREAD

**Issue by tklengyel** [tklengyel/drakvuf#936]
Currently the injector code is extremely hard to follow with multiple injection methods all being jammed into `injector_int3_cb` for Windows. Splitting each injection method into its own file and own callback function would make the various paths a lot easier to follow. An attempt at this has been already made in https://github.com/tklengyel/drakvuf/tree/injector but it needs to be dusted off and restarted on the latest `libinjector`.

**Comment by icedevml** [comment]
Btw. we also have a lot of continations in this code. I.e. the situations where you need to return from the function and it would re-enter shortly. This is the common problem in the whole JavaScript world and they've solved it with `async`/`await`.

I think the coroutine paradigim would be super helpful in the injector and also in a few other places where continuations are used. Unfortunately this is yet to come with C++20 :(
